src/decision-engine/index.js

```diff
@@ -4,19 +4,13 @@ const each = require('async/each')
 const eachSeries = require('async/eachSeries')
 const waterfall = require('async/waterfall')
 const setImmediate = require('async/setImmediate')
-
 const map = require('async/map')
-const debounce = require('lodash.debounce')
-const uniqWith = require('lodash.uniqwith')
-const find = require('lodash.find')
-const values = require('lodash.values')
-const groupBy = require('lodash.groupby')
-const pullAllWith = require('lodash.pullallwith')
+const debounce = require('just-debounce-it')
 
 const Message = require('../types/message')
 const Wantlist = require('../types/wantlist')
 const Ledger = require('./ledger')
-const logger = require('../utils').logger
+const { logger, groupBy, pullAllWith, uniqWith } = require('../utils')
```

Comment:
lodash methods hide to much magic look at the OP example i couldn't replace isEqualWith because it does to much stuff and our logic is completely dependent on it (i really think we should refactor that).
if you look at the implementations they are really simple and more functional as they should be.

Reply:
plus the size gains are huge the OP show only 10% because bignumber is bigger than big and makes up for most of the lodash reduction